### ISSUE TITLE
new folder pair should show both local and remote sub-directories, if existing

### ISSUE THREAD

**Issue by martin-rueegg** [owncloud/client#4278]
when creating a new folder sync, and on both sides sub-folders do already exist, only the remote folders are shown and (un-)selectable, while the local folders are not shown (yet synchronized).

local folders should also show up in the list.

not sure, how the size indication should be handled. maybe two sizes could be given, in case that the remote and local size differ.

(client: Version 2.1.0 (build 5683). Windows.)

**Comment by guruz** [comment]
FYI @MTRichards as this comes up again and again :)

**Comment by martin-rueegg** [comment]
@guruz, i'm sorry if i'm generating duplicates! i tried to use the search function, but it's far from easy to find the right thing (mostly i find others ... :-D)

**Comment by MTRichards** [comment]
Let me see if I understand.

When setting up to sync a new folder from the desktop, and that local folder has sub folders, all client side sub-folders are synced by default to the server. While you can selectively sync server side, you can't selectively sync client side.

Is that the issue?

**Comment by guruz** [comment]
(e.g. https://github.com/owncloud/client/issues/3823 might be what @martin-rueegg means..?)

**Comment by martin-rueegg** [comment]
yep, @MTRichards.

(with the exception, that when the local folder also exists on the server already, i can unselect it, of course.)

client | server | list
------|------|-----------
A|| -
B|B|[x]
 |C|[x]
D|D'|[x]

`client`: local folder on client
`server`: remote folder (on server)
`list`: does it appear in the folder selection list? `[x]` = yes, `-` = no.

in case D where the content of the two corresponding folders is not (yet) in sync, it shows only the size of the remote folder.

**Comment by martin-rueegg** [comment]
right, seems to be a duplicate of #3823 indeed, @guruz. only that in my case i actually want to upload the local data. i just realized, if i wouldn't (like @despens), then i would have a problem ...

**Comment by MTRichards** [comment]
The other problem I see is that if you link the account (add the folder and sub folders locally) and then sync to the server the local sub-folders, even if you don't want to - when you unselect the local sub folders they get deleted locally.

A lot of this stems from the idea that ownCloud "owns" the local sync folder, so the idea of adding ownCloud to an existing folder structure means you are making ownCloud now "own" the folder and all sub-folders. I can see that one might not want to do that, but not how to fix the problem.

For example, if I allow folders inside a local sync folder to not sync with the server, but they are still there, then what happens?

**Comment by martin-rueegg** [comment]
i think there are two separate issues, even though they are somewhat interconnected.

the workflow laid out by @MTRichards would suggest, that i first establish a sync folder definition (btw. what is the correct term for this? other apps use the term *folder pair*, as the local and remote folder gets paired, but here the term *pair* seems unused), then - once established - i start placing files into that folder locally (whether or not remote folders have been downloaded or not).

the UI follows this workflow by not showing any local folders. yet the sync process does *not* follow this workflow, as it *does* upload only locally existing existing folders.

thus, this issue here is more intended to solve this discrepancy. maybe until there is a general consent about the handling of *blacklisted* or *ignored* folders is established, the local folders - if unselected - would simply be added to the ignore-list and removed from the UI. leaving them selected instead would immediately create the folders on the server (without uploading the files yet), so that the situation is consistent with expected behaviour when the local folder was empty in the first place.
